android/titanium/src/java/org/appcelerator/titanium/TiBaseActivity.java

```diff
@@ -1724,7 +1713,11 @@ protected void fireOnDestroy()
 
 	protected boolean shouldFinishRootActivity()
 	{
-		return canFinishRoot && getIntentBoolean(TiC.INTENT_PROPERTY_FINISH_ROOT, false);
+		boolean isIntentRequestingFinishRoot = false;
```

Comment:
you could check and exit early `if (!TiBaseActivity.canFinishRoot)` to avoid the "work" of checking null/grabbing the boolean.